### PR TITLE
Update Rector to use PHP 8.1 rulesets.

### PR DESCRIPTION
**Pull request by markdorison**
## Description
Update Rector to use PHP 8.1 rulesets.

## Motivation / Context
Follow up to 8f91336ae80854217d66e3ba4f1c3903d63e8e82.

## Testing Instructions / How This Has Been Tested
Tests pass.

### CODE REVIEW

File: src/Factory.php [hunk 1]
```diff
@@ -24,13 +24,6 @@ class Factory
     /**
      * Constructor.
      *
-     * @param array $config
-     *   An array of configurations. You need at least the 'key'.
```
Comment (adamzimmermann) [inline src/Factory.php:28]: This seems like it could be valuable?

File: src/Endpoints/OAuth2.php [hunk 2]
```diff
@@ -16,11 +16,8 @@ class OAuth2 extends Endpoint
     /**
      * Get OAuth 2.0 Access Token.
      *
-     * @param string $clientId
      *   The Client ID of your app.
```
Comment (adamzimmermann) [inline src/Endpoints/OAuth2.php]: We keep these lines but remove the `@param` lines? Is that intentional?

File: src/Endpoints/MediaFile.php [hunk 3]
```diff
@@ -14,11 +14,6 @@ class MediaFile extends Endpoint
     /**
      * Retrieve subtitles for an item.
      *
-     * @param string $video_id
-     *   The item's record ID.
```
Comment (adamzimmermann) [inline src/Endpoints/MediaFile.php:18]: This seems like it could be valuable context to know that the video ID is a Record ID.

**Review by adamzimmermann — COMMENTED** [review]
While I love removing comments like these most of the time, I worry that some of these actually have valuable information. Thoughts?

File: src/Endpoints/MediaFile.php (hunk 3 above)
Comment (markdorison) [reply src/Endpoints/MediaFile.php:18]: Could I rename the variable and achieve clarity that way?

File: src/Endpoints/OAuth2.php (hunk 2 above)
Comment (markdorison) [reply src/Endpoints/OAuth2.php]: No, I missed this one. Will update.

File: src/Factory.php (hunk 1 above)
Comment (markdorison) [reply src/Factory.php:28]: Added comment below. We could also check and throw an exception if it isn't there. What do you think?

File: src/Endpoints/MediaFile.php (hunk 3 above)
Comment (markdorison) [reply src/Endpoints/MediaFile.php:18]: See below.

**Review by adamzimmermann — APPROVED** [review]

**Pull request merged**